packages/web3-validator/src/utils.ts

```diff
@@ -205,37 +205,43 @@ export const abiSchemaToJsonSchema = (
 			nestedTuple.$id = abiName;
 			(lastSchema.items as JsonSchema[]).push(nestedTuple);
 		} else if (baseType === 'tuple' && isArray) {
-			const arraySize = arraySizes[0];
-			const item: JsonSchema = {
-				$id: abiName,
-				type: 'array',
-				items: abiSchemaToJsonSchema(abiComponents, abiName),
-				maxItems: arraySize,
-				minItems: arraySize,
-			};
-
-			if (arraySize < 0) {
-				delete item.maxItems;
-				delete item.minItems;
-			}
-
-			(lastSchema.items as JsonSchema[]).push(item);
+		    // iterate over arraySizes array to each dimension
+		    for (let i = 0; i < arraySizes.length; i++) {
```

Comment:
@EtlesL 
some of fixes are required for Linter to pass https://github.com/web3/web3.js/actions/runs/8058948881/job/22029093796?pr=6836#step:5:246 , also could you add unit tests for this usecase, Thanks for your contribution,